### ISSUE TITLE
shader kills alpha (of PNG's) when using filter(shader) ONLY in 3.4

### ISSUE THREAD

**Issue by hype** [processing/processing#5650]
howdy, @codeanticode 

issue occurs ONLY in 3.4... when using any `PShader` + transparent PNG's

this code works as expected: when using the shader() call

```
void draw() {
  background(#FF3300);
  shader(d1);
  image(img, 0, 0);
}
```
![screen shot 2018-09-14 at 2 13 16 pm](https://user-images.githubusercontent.com/139746/45567638-5a6b8a80-b828-11e8-9c26-81567bb300f5.png)

but if you do this... using the `filter(shader)` call... the transparency of the PNG does not behave as expected.

```
void draw() {
  background(#FF3300);
  filter(d1);
  image(img, 0, 0);
}
```
![screen shot 2018-09-14 at 2 14 57 pm](https://user-images.githubusercontent.com/139746/45567723-956dbe00-b828-11e8-9de9-e60a12a3bbbb.png)

in most of current code examples I enjoy the flexibility of drawing with a `PGraphics` and using `pgraphics.filter(shader)` to apply what shader gets applied where... as seen below :

![screen shot 2018-09-14 at 2 06 28 pm](https://user-images.githubusercontent.com/139746/45567856-ee3d5680-b828-11e8-9534-b59eafa3cb8e.png)

**Comment by Sardtok** [comment]
This is fixed in #5647. filter(PShader) temporarily sets the blend mode to REPLACE, which would disallow changing the blend mode back in 3.4.

**Comment by hype** [comment]
cheers. @Sardtok I always try to do a thorough search to make sure I'm not double posting... sorry for not catching this.
